spec/integration/util/windows/monkey_patches/dir_spec.rb

```diff
@@ -0,0 +1,231 @@
+      expect { Dir.read_junction([]) }.to raise_error(TypeError)
+    end
+  end
+
+  describe '.junction?' do
+    it 'returns a boolean value' do
+      expect(Dir).to respond_to(:junction?)
+      expect { Dir.create_junction(ascii_to, from) }.not_to raise_error
+      expect(Dir.junction?(from)).to be false
+      expect(Dir.junction?(ascii_to)).to be true
+      expect(Dir.junction?(Pathname.new(ascii_to))).to be true
+    end
+  end
+
+  describe '.reparse_dir?' do
+    it 'aliases junction' do
+      expect(Dir).to respond_to(:reparse_dir?)
+      expect(Dir.method(:reparse_dir?)).to eq(Dir.method(:junction?))
+    end
+  end
+
+  describe '.empty?' do
+    it 'returns expected result' do
+      expect(Dir).to respond_to(:empty?)
+      expect(Dir.empty?("C:\\")).to be false
+      expect(Dir.empty?(from)).to be true
+      expect(Dir.empty?(Pathname.new(from))).to be true
+    end
+  end
+
+  describe '.pwd' do
+    it 'has basic functionality' do
+      expect(Dir).to respond_to(:pwd)
+      expect { Dir.pwd }.not_to raise_error
+      expect(Dir.pwd).to be_a_kind_of(String)
+    end
+
+    it 'aliases getwd' do
+      expect(Dir.method(:pwd)).to eq(Dir.method(:getwd))
+    end
+
+    it 'returns full path even if short path was just used' do
+      Dir.chdir("C:\\Progra~1")
+      expect(Dir.pwd).to eq("C:\\Program Files")
+    end
+
+    it 'returns full path even if long path was just used' do
+      Dir.chdir("C:\\Program Files")
+      expect(Dir.pwd).to eq("C:\\Program Files")
+    end
+
+    it 'uses standard case conventions' do
+      Dir.chdir("C:\\PROGRAM FILES")
+      expect(Dir.pwd).to eq("C:\\Program Files")
+    end
+
+    it 'converts forward slashes to backslashes' do
+      Dir.chdir("C:/Program Files")
+      expect(Dir.pwd).to eq("C:\\Program Files")
+    end
+  end
+
+  describe 'constants' do
+    ["DESKTOP", "INTERNET", "PROGRAMS", "CONTROLS",
+     "PRINTERS", "PERSONAL", "FAVORITES", "STARTUP", "RECENT", "SENDTO",
+     "BITBUCKET", "STARTMENU", "MYDOCUMENTS", "MYMUSIC", "MYVIDEO",
+     "DESKTOPDIRECTORY", "DRIVES", "NETWORK", "NETHOOD", "FONTS",
+     "TEMPLATES", "COMMON_STARTMENU", "COMMON_PROGRAMS",
+     "COMMON_STARTUP", "COMMON_FAVORITES", "COMMON_DESKTOPDIRECTORY",
+     "APPDATA", "PRINTHOOD", "LOCAL_APPDATA", "ALTSTARTUP",
+     "COMMON_ALTSTARTUP", "INTERNET_CACHE", "COOKIES", "HISTORY",
+     "COMMON_APPDATA", "WINDOWS", "SYSTEM", "PROGRAM_FILES",
+     "MYPICTURES", "PROFILE", "SYSTEMX86", "PROGRAM_FILESX86",
+     "PROGRAM_FILES_COMMON", "PROGRAM_FILES_COMMONX86",
+     "COMMON_TEMPLATES", "COMMON_DOCUMENTS", "CONNECTIONS",
+     "COMMON_MUSIC", "COMMON_PICTURES", "COMMON_VIDEO", "RESOURCES",
+     "RESOURCES_LOCALIZED", "COMMON_OEM_LINKS", "CDBURN_AREA",
+     "COMMON_ADMINTOOLS", "ADMINTOOLS"].each do |constant|
+      it "#{constant} is set" do
+        expect(Dir.const_get(constant)).not_to be_nil
+        expect(Dir.const_get(constant)).to be_a_kind_of(String)
+      end
```

Comment:
This will raise error if a constant is not defined.
Would be better to check something like?
```ruby
expect { Dir.const_get(constant) }.not_to raise_error
```